### ISSUE TITLE
train.py

### ISSUE THREAD

**Issue by rui99999** [feiyuhuahuo/Anomaly_Prediction#11]
When I run python train.py --dataset=avenue,The following problems arise
![image](https://user-images.githubusercontent.com/53531539/96702871-c5eb9300-13c4-11eb-8dac-024f5edbce78.png)

**Comment by feiyuhuahuo** [comment]
What's your PyTorch version? Have you modify the code?

**Comment by rui99999** [comment]
> What's your PyTorch version? Have you modify the code?

PyTorch version 1.6.0 ; dataset location and batch_size=8

**Comment by feiyuhuahuo** [comment]
I need to test on PyTorch 1.6 and check the code. I can't tell whether it's an error now.

**Comment by rui99999** [comment]
I downgraded pytorch to 1.4.0 and solved this problem.Thank you for your reply

**Comment by feiyuhuahuo** [comment]
@rui99999 
I updated this project. It's compatible with PyTorch 1.5 ~1.6 now.

**Comment by sthoduka** [comment]
I had the same error and it was fixed by moving the optimizer step after running both backward passes. 
i.e. in train.py (around line 130):

```
            optimizer_D.zero_grad()
            D_l.backward()

            # Train generator
            optimizer_G.zero_grad()
            G_l_t.backward()

            optimizer_D.step()
            optimizer_G.step()
```

got a hint from [here](https://discuss.pytorch.org/t/backward-one-of-the-variables-needed-for-gradient-computation-has-been-modified-by-an-inplace-operation/102126)
